Objects/bytearrayobject.c

```diff
@@ -1729,6 +1729,11 @@ bytearray_extend(PyByteArrayObject *self, PyObject *iterable_of_ints)
 
     while ((item = PyIter_Next(it)) != NULL) {
         if (! _getbytevalue(item, &value)) {
+            if (PyErr_ExceptionMatches(PyExc_TypeError) && PyUnicode_Check(iterable_of_ints)) {
+                PyErr_Format(PyExc_TypeError,
+                             "'%.100s' object should be encoded",
+                             Py_TYPE(iterable_of_ints)->tp_name);
```

Comment:
If one passes a string literal, one should just add the `b` prefix insteand of `.encode()`.

```suggestion
                             "str object should be bytes");
```
The message is the point of his suggestion.  I am presuming that a dummy arg for the constant string is not needed.  In any case, bytes are not, in general, encoded text, although attempting to pass a str is likely to indicate that as the intended interpretation.  A bytearray would also be OK, but anyone using one would likely know that.

Reply:
Maybe quote typenames?

Reply:
A different wording was chosen.